src/histogram.rs

```diff
@@ -0,0 +1,28 @@
+
+use std::collections::BTreeMap;
+
+pub struct Histogram<K: Ord + Clone> {
```

Comment:
Nice one Peter, strictly speaking a histogram collects items in defined buckets; so is not a histogram in the strict sense of the word.  Hence I called it 'frequency count' which arguably is not a good name either.  Nice call regardless of semantics to abstract these functions !